sklearn/metrics/pairwise.py

```diff
@@ -1566,6 +1566,13 @@ def additive_chi2_kernel(X, Y=None):
     return result
 
 
+@validate_params(
+    {
+        "X": ["array-like", "sparse matrix"],
+        "Y": ["array-like", "sparse matrix", None],
+        "gamma": [Interval(Real, None, None, closed="neither")],
```

Comment:
I am also surprised here that it does fail because we don't allow `ndarray` which would be required by Gaussian processes.

Reply:
Indeed, here we would trigger a regression:

```python
In [2]: from sklearn.datasets import make_friedman2
   ...: from sklearn.gaussian_process import GaussianProcessRegressor
   ...: from sklearn.gaussian_process.kernels import DotProduct, WhiteKernel, PairwiseKernel
   ...: X, y = make_friedman2(n_samples=500, noise=0, random_state=0)
   ...: kernel = DotProduct() + WhiteKernel() + PairwiseKernel(gamma=1.0, metric="chi2")
   ...: gpr = GaussianProcessRegressor(kernel=kernel,
   ...:         random_state=0).fit(X, y)
```

```
InvalidParameterError: The 'gamma' parameter of chi2_kernel must be a float in the range (-inf, inf). Got array([1.]) instead.
```

So it would be useful to add in `test_gpr.py` this minimal example to be sure that we have a test triggering this issue.

Here, it means that we need:

```suggestion
        "gamma": [Interval(Real, None, None, closed="neither"), Hidden(np.ndarray)],
```

Reply:
@rand0wn please also add a test with what @glemaitre points out. Also, why don't we want to document `gamma` as an array publicly?

Reply:
Because it is an array with unique values. We found out that the only case that this happens is internal to the GP models that provide such entry.

We therefore should still accept this type of entry to avoid regression but we should not document it since we don't support `gamma` being several values in an array.